spark/src/main/scala/ai/chronon/spark/TableUtils.scala

```diff
@@ -250,18 +250,14 @@ case class TableUtils(sparkSession: SparkSession) {
       sparkSession.sql(s"SELECT * FROM $tableName where $partitionColumn='$partitionFilter' LIMIT 1").collect()
       true
     } catch {
-      case e: RuntimeException =>
+      case e: Exception =>
         if (e.getMessage.contains("ACCESS DENIED"))
           logger.error(s"[Error] No access to table: $tableName ")
         else {
           logger.error(s"[Error] Encountered exception when reading table: $tableName.")
           e.printStackTrace()
         }
         false
-      case ex: Exception =>
```

Comment:
Ideally we should figure out why the error message does not fall into first case "ACCESS DENIED" since it do have the target string, and handle it properly. 

Remove the last case would make permission check more aggressive - any exception we encountered would result in a "No permission" which might not be accurate.

Reply:
@SophieYu41 I updated the code. The issue is because the exception is recognized as sparkException instead of java runtime exception, it does include "ACCESS DENIED" in error message. 

One more question, in which case we still want the analyzer succeed with exception caught here? No matter what type of exception it is we still need to remind users reading from tables failed right?

Reply:
Great! Thanks!

> in which case we still want the analyzer succeed with exception caught here?
Think it is about unknown of the unknowns.  In case there is some unexpected exception/runtimeException stopping us evaluating the table permission, the question is if we would like to hard-stop the backfill. 
I would prefer to not since real permission issue would surface during backfill anyways, the check here is to fail fast. But if we have false negative, backfill will fail even there is no permission issue. Feels like less harmful to have false-positive rather than false negative.

Reply:
We will hard stop regardless no?

Reply:
@nikhilsimha we would hard stop if any validations failed. Think the questions here is - do we want to hard stop if we encountered errors and not able to verify if table permission is present or not.